molecule/default/tests/test_default.py

```diff
@@ -56,15 +46,33 @@ def test_redhat_updated_time(host):
         assert (datetime.datetime.now() - last_update).total_seconds() <= 20 * 60
 
 
-def test_redhat_updated_command_output(host):
-    """Test that RedHat instances were updated."""
-    if (
-        host.system_info.distribution == "amzn"
-        or host.system_info.distribution == "fedora"
-    ):
+def test_yum_updated_command_output(host):
+    """Test that yum-based instances were updated."""
+    if host.system_info.distribution in ["amzn"]:
         yum_output = host.run("yum update")
         # If the update succeeded or there was nothing to update
-        assert (
-            "No packages marked for update" in yum_output.stdout
-            or "Nothing to do" in yum_output.stdout
+        assert "No packages marked for update" in yum_output.stdout
+
+
+# This test can fail if there were no updates to install.
+@pytest.mark.xfail
+def test_dnf_updated_time(host):
+    """Test that dnf-based instances were updated."""
+    if host.system_info.distribution in ["fedora"]:
+        last_update = datetime.datetime.strptime(
+            host.run(
+                "yum --quiet history list | cut --delimiter='|' --fields=3-4 | grep --fixed-strings U | cut --delimiter='|' --fields=1 | head --lines=1"
+            ).stdout.strip(),
+            "%Y-%m-%d %H:%M",
         )
+        # Make sure that the instance was updated in the last 20
+        # minutes.
+        assert (datetime.datetime.now() - last_update).total_seconds() <= 20 * 60
+
+
+def test_dnf_updated_command_output(host):
+    """Test that dnf-based instances were updated."""
+    if host.system_info.distribution in ["fedora"]:
+        dnf_output = host.run("dnf update")
+        # If the update succeeded or there was nothing to update
```

Comment:
I believe this should be changed to align with the assertion.

Reply:
Please see commit d2c0824.